### ISSUE TITLE
ttnn.softmax low pcc when input is -338953138925153547590470800371487866880.00000

### ISSUE THREAD

**Issue by amalbasaTT** [tenstorrent/tt-metal#19781]
**Describe the bug**
ttnn.softmax is one of the causes for low PCC in CLIP model.

Input spec:
- inputs are all -338953138925153547590470800371487866880.00000
- input shape: [16, 7, 7]
- datatype: ttnn.bfloat16
- layout: ttnn.TILE_LAYOUT
- dim: -1
- numeric_stable: True

PCC is 0.0.
This spec is found in CLIP model and is one of the causes of low accuracy in that model.

**To Reproduce**
Run unit test:
```
from loguru import logger
import random
import pytest
import torch
import ttnn
import traceback
from models.utility_functions import torch_random
from tests.ttnn.utils_for_testing import assert_with_pcc
aten = torch.ops.aten


def max_relative_error(expected, calculated):
    assert expected.shape == calculated.shape, "Tensors must have the same shape"
    if expected.dtype == torch.bool:
        expected = expected.to(torch.int)
    if calculated.dtype == torch.bool:
        calculated = calculated.to(torch.int)
    relative_error = torch.abs(expected - calculated) / (torch.abs(expected) + 1e-8)
    mre = torch.max(relative_error).item()
    inds = torch.where(relative_error == mre)
    return mre, inds


def run_softmax_tests(
    input_shape, 
    dtype, 
    dlayout,
    dim,
    numeric_stable,
    device,
):

    x = torch.full(size=input_shape[0], fill_value=-338953138925153547590470800371487866880.00000, dtype=torch.bfloat16)

    try:
        ref_value = aten._softmax.default(x, dim, False)
        
        tt_x = ttnn.from_torch(x, dtype=dtype[0], layout=dlayout[0], device=device)

        tt_result = ttnn.softmax(tt_x, dim, numeric_stable=numeric_stable)
        tt_result = ttnn.to_torch(tt_result)
    except Exception as e:
        logger.warning(f"Test execution crashed: {e}")
        print(traceback.format_exc())
        raise e

    assert len(tt_result.shape) == len(ref_value.shape)
    assert tt_result.shape == ref_value.shape
    assert_with_pcc(ref_value, tt_result, 0.999)


test_sweep_args = [
    (
        [[16, 7, 7]],
        [ttnn.bfloat16],
        [ttnn.TILE_LAYOUT],
        -1, 
        True,
    ),
]


@pytest.mark.parametrize(
    "input_shape, dtype, dlayout, dim, numeric_stable",
    (test_sweep_args),
)
def test_softmax(input_shape, dtype, dlayout, dim, numeric_stable, device):
    run_softmax_tests(input_shape, dtype, dlayout, dim, numeric_stable, device)

```

**Comment by bbradelTT** [comment]
This is related to implicit padding.

We'll add that into the op itself.

For now as a workaround, before calling ttnn.softmax, you can do that directly via
```
tt_x = ttnn.fill_implicit_tile_padding(tt_x, -float("inf"))
```

**Comment by nsorabaTT** [comment]
PR https://github.com/tenstorrent/tt-metal/pull/20008 on the way to address this issue.

**Comment by nsorabaTT** [comment]
Softmax kernel is using FF00 as padding instead of FFF0 (-inf in bfloat16). Fixing the padding makes this test pass. Making a fix after accounting for padding in fp32 and bfloat8_b formats.

**Comment by amalbasaTT** [comment]
@nsorabaTT will that PR be merged soon?

**Comment by nsorabaTT** [comment]
> [@nsorabaTT](https://github.com/nsorabaTT) will that PR be merged soon?

Hi @amalbasaTT Yes, I am looking into the perf impact and also making sure the fix addresses FP32 and BFLOAT8_b datatypes as well. PR will be merged by tomorrow.

**Comment by nsorabaTT** [comment]
@amalbasaTT PR is merged. Please let me know if you see any issues.
